storage/src/main/scala/ch/epfl/bluebrain/nexus/storage/routes/StorageRoutes.scala

```diff
@@ -33,38 +34,54 @@ class StorageRoutes()(implicit storages: Storages[AkkaSource], hc: HttpConfig) {
-                    // Upload file
-                    fileUpload("file") { case (_, source) =>
-                      complete(Created -> storages.createFile(name, path, source).unsafeToFuture())
-                    }
-                  }
-                },
-                put {
-                  // Link file/dir
-                  entity(as[LinkFile]) { case LinkFile(source) =>
-                    validatePath(name, source) {
-                      complete(storages.moveFile(name, source, path).runWithStatus(OK))
+        pathPrefix("files") {
+          bucketExists(name).apply { implicit bucketExistsEvidence =>
+            concat(
+              extractPath(name) { path =>
+                operationName(s"/${hc.prefix}/buckets/{}/files/{}") {
+                  concat(
+                    put {
+                      pathNotExists(name, path).apply { implicit pathNotExistEvidence =>
+                        // Upload file
+                        fileUpload("file") { case (_, source) =>
+                          complete(Created -> storages.createFile(name, path, source).unsafeToFuture())
+                        }
+                      }
+                    },
+                    put {
+                      // Link file/dir
+                      entity(as[LinkFile]) { case LinkFile(source) =>
+                        validatePath(name, source) {
+                          complete(storages.moveFile(name, source, path).runWithStatus(OK))
+                        }
+                      }
+                    },
+                    // Get file
+                    get {
+                      pathExists(name, path).apply { implicit pathExistsEvidence =>
+                        storages.getFile(name, path) match {
+                          case Right((source, Some(_))) => complete(HttpEntity(`application/octet-stream`, source))
+                          case Right((source, None))    => complete(HttpEntity(`application/x-tar`, source))
+                          case Left(err)                => complete(err)
+                        }
+                      }
                     }
-                  }
-                },
-                // Get file
-                get {
-                  pathExists(name, path).apply { implicit pathExistsEvidence =>
-                    storages.getFile(name, path) match {
-                      case Right((source, Some(_))) => complete(HttpEntity(`application/octet-stream`, source))
-                      case Right((source, None))    => complete(HttpEntity(`application/x-tar`, source))
-                      case Left(err)                => complete(err)
+                  )
+                }
+              },
+              operationName(s"/${hc.prefix}/buckets/{}/files") {
+                post {
+                  // Copy files within protected directory
+                  entity(as[NonEmptyList[CopyFile]]) { files =>
```

Comment:
The thing that bugs me is that copying a lot of files complicates things as it makes it more likely to hit some timeouts

Reply:
Hm yeah we could keep track of this and add a limit in delta if it's a problem - do we have metrics for individual endpoints? If not I could add logging